import "ydb/library/actors/protos/actors.proto";
import "ydb/core/tx/columnshard/data_sharing/protos/links.proto";
import "ydb/core/tx/columnshard/data_sharing/protos/data.proto";
import "ydb/core/tx/columnshard/data_sharing/protos/sessions.proto";
import "ydb/core/tx/columnshard/data_sharing/protos/initiator.proto";

package NKikimrColumnShardDataSharingProto;

message TEvConfirmFromInitiator {
    optional string SessionId = 1;
}

message TEvAckFinishFromInitiator {
    optional string SessionId = 1;
}

message TEvStartToSource {
    optional TSourceSession Session = 1;
}

message TEvSendDataFromSource {
    optional string SessionId = 1;
    optional uint64 PackIdx = 2;
    repeated TPathIdData PathIdData = 3;
    optional uint64 SourceTabletId = 4;
}

message TEvAckDataToSource {
    optional uint64 PackIdx = 1;
    optional string SessionId = 2;
}

message TEvFinishedFromSource {
    optional string SessionId = 1;
    optional uint64 SourceTabletId = 2;
}

message TEvAckFinishToSource {
    optional string SessionId = 1;
}

message TEvCheckStatusFromInitiator {
    optional string SessionId = 1;
}

message TEvCheckStatusResult {
    optional string ClassName = 1;
    optional string SessionId = 2;
    optional TInitiator.TStatus Status = 3;
}

message TEvApplyLinksModification {
    optional uint64 InitiatorTabletId = 1;
    optional string SessionId = 2;
    optional uint64 PackIdx = 3;
    optional TTaskForTablet Task = 4;
}

message TEvApplyLinksModificationFinished {
    optional uint64 ModifiedTabletId = 1;
    optional string SessionId = 2;
    optional uint64 PackIdx = 3;
}